### PR TITLE
chore(Rust): Implement changes outlined in #3560

### LINKED ISSUE #3560: Refactor `configuration set-default-node` command to stop mapping errors manually
Command: [`configuration set-default-node`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/configuration/set_default_node.rs)

## Current

The current implementation is calling `std::process::exit`, which is something we want to stop using to handle errors properly and give the user a better-formatted output when a command fails. 

## Desired

- `run` method (main entry point of the subcommand) should contain a single line of code calling a function (run_impl/rpc/...) that returns a `crate::Result<()>`
- `std::process::exit` should not be used manually to map the errors of the command's logic
- `run` method will return `()`, and will unwrap the result like it's done [here](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/node/delete.rs#L24) (note that there is a single call to `std::process::exit` at the command's `run` method and with the error `code` derived from the error returned by the `run_impl` method)

You can take the [`node delete`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/node/delete.rs#L23) command as a reference.

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

### PR DESCRIPTION
**Pull request by AntoniosBarotsis**
Closes #3560

Decouples [`set_default_node`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/configuration/set_default_node.rs)'s implementation from the `run` method for easier error handling.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@AntoniosBarotsis Thank you for spending time on this, before we can merge please

- [x] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

**Comment by AntoniosBarotsis** [comment]
Just made [the PR](https://github.com/build-trust/ockam-contributors/pull/116) :)

**Comment by mrinalwadhwa** [comment]
@AntoniosBarotsis I pushed some commit message updates to your branch to satisfy our commit message linter. It looks like Clippy is failing on the PR, could you please resolve those errors before we merge https://github.com/build-trust/ockam/actions/runs/3184031888/jobs/5191968966

**Comment by AntoniosBarotsis** [comment]
It seems that I am not getting the error locally due to some compilation issues as I am on Windows, will fix but make sure to check it again just to be sure it's fine!

**Comment by mrinalwadhwa** [comment]
@AntoniosBarotsis linters are happy now. The rust nightly checks can be ignored.
We can merge after @adrianbenavides has a change to review the change in his morning.

Thank you for an awesome first contribution 🥳

File: implementations/rust/ockam/ockam_command/src/configuration/set_default_node.rs [hunk 1]
```diff
@@ -12,16 +13,30 @@ pub struct SetDefaultNodeCommand {
...
+                exitcode::IOERR,
+                anyhow!(format!("failed to update configuration: {}", e)),
+            ));
+        }
+    } else {
+        return Err(crate::error::Error::new(
+            exitcode::CANTCREAT,
+            anyhow!(format!("Node ({}) is not registered yet", name)),
+        ));
+    }
```
Comment (adrianbenavides) [inline implementations/rust/ockam/ockam_command/src/configuration/set_default_node.rs]: You can simplify this if you don't map the errors manually:

```rust
options.config.get_node(name)?;
options
    .config
    .set_default_node(&get_final_element(name).to_owned());
options.config.persist_config_updates()?;
```

**Review by adrianbenavides — CHANGES_REQUESTED** [review]
Hey @AntoniosBarotsis, it looks pretty great. Just a small change and will be ready to merge!

**Comment by adrianbenavides** [comment]
@mergifyio rebase

**Review by adrianbenavides — DISMISSED** [review]
Ready to merge @AntoniosBarotsis , thank you so much for your PR! :metal:

**Comment by adrianbenavides** [comment]
@mergifyio rebase

**Review by adrianbenavides — DISMISSED** [review]

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**